src/main/java/com/aws/greengrass/easysetup/DeviceProvisioningHelper.java

```diff
@@ -415,9 +447,10 @@ private Optional<String> getPolicyArn(String policyName, Region awsRegion) {
         String partition = awsRegion.metadata().partition().id();
         try {
             // Check if a managed policy exists with the name
-            return Optional.of(iamClient.getPolicy(software.amazon.awssdk.services.iam.model.GetPolicyRequest.builder()
-                    .policyArn(String.format(MANAGED_IAM_POLICY_ARN_FORMAT, partition, policyName)).build()).policy()
-                    .arn());
+            return Optional.of(iamClient
+                    .getPolicy(software.amazon.awssdk.services.iam.model.GetPolicyRequest.builder()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Potential log injection detected. Ensure all untrusted input is properly sanitized before logging. Use parameterized logging or validate input against a allow list to prevent log injection vulnerabilities. Consider using a dedicated logging library&#39;s built-in sanitization features when available. Learn more - https://cwe.mitre.org/data/definitions/117.html